lib/sparrow/fcm/v1/notification.ex

```diff
@@ -124,10 +124,32 @@ defmodule Sparrow.FCM.V1.Notification do
         {:error, :invalid_notification}
 
       true ->
-        %{notification | data: Map.new(data)}
+        v1 = %{notification | data: Map.new(data)}
```

Comment:
How about using `with` cause and calling each module with their struct directly? E.g.

```
with {:ok, notification} <- verify(notification),
        {:ok, android} <- Sparrow.FCM.V1.Android.verify(notification.android),
        {:ok, webpush} <- Sparrow.FCM.V1.Android.verify(notification.webpush) do
... update notification with new android and webpush sub-structs ...
```

Reply:
Also, I would change the name of function `verify`, as it actually mutates the data. I don't know how, but `verify` feels misleading.

Reply:
`normalize`? `cast`?